cmd/podman/cp.go

```diff
@@ -223,8 +228,23 @@ func copy(src, destPath, dest string, idMappingOpts storage.IDMappingOptions, ch
 	copyWithTar := chrootarchive.CopyWithTarAndChown(chownOpts, digest.Canonical.Digester().Hash(), idMappingOpts.UIDMap, idMappingOpts.GIDMap)
 	untarPath := chrootarchive.UntarPathAndChown(chownOpts, digest.Canonical.Digester().Hash(), idMappingOpts.UIDMap, idMappingOpts.GIDMap)
 
-	if srcfi.IsDir() {
+	filename := filepath.Base(destPath)
+	if filename == "-" && !isFromHostToCtr {
```

Comment:
Wait, how does this work? We never pass the `"-"` into this function, it's reset to `os.Stdin.Name()` above?

Reply:
the above `-` reset to `os.Stdin.Name()` as a source is used for copying tar from my machine to a container.
this function receives `-` from destPath when streaming a tar archive from a container to host